src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -444,5 +444,635 @@ public void M(int x)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (1)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b')
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Jump if False (Regular) to Block[B3]
+        IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'a')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b')
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c')
+          Value: 
+            IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: System.String) (Syntax: 'c')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'b = $""{(a ? b : c)}"";')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.String) (Syntax: 'b = $""{(a ? b : c)}""')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'b')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String) (Syntax: '$""{(a ? b : c)}""')
+                  Parts(1):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{(a ? b : c)}')
+                        Expression: 
+                          IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'a ? b : c')
+                        Alignment: 
+                          null
+                        FormatString: 
+                          null
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void InterpolatedStringExpression_TextAndInterpolationParts_Flow()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M(bool a, string b, string c)
+    /*<bind>*/{
+        b = $""String1 {(b = a ? b : c)} and String2 {(a ? b : c)}"";
```

Comment:
>b = a ? b : c [](start = 25, length = 13)

Is there anything interesting about an assignment in this position? If not, I recommend removing the assignment. If that scenario is covered by another test, removing this test.

Reply:
Yes, there is an additional flow l-value flow capture in an interpolation, which none of the other tests cover.

---
In reply to: [185917281](https://github.com/dotnet/roslyn/pull/26545#discussion_r185917281) [](ancestors = 185917281)

Reply:
> Yes, there is an additional flow l-value flow capture in an interpolation, which none of the other tests cover.

How is this interesting? Is there some kind of unique interaction going on here?

---
In reply to: [185918175](https://github.com/dotnet/roslyn/pull/26545#discussion_r185918175) [](ancestors = 185918175,185917281)

Reply:
The unit test is testing "a write in an interpolation between two reads of the same variable in other interpolations within the same interpolation string". It may not be interesting based on our current CFG implementation as our flow captures are not optimized. If we chose to run an optimization pass over the generated flow captures, then we can remove lot of redundant flow captures of same value when there is no write between 2 reads. In that scenario, presence of a write between two reads is indeed a very interesting scenario as the two reads need to be different flow captures. Regardless, I felt this was an interesting scenario mix to cover. If you insist on not involving any such hybrid scenarios, I can reduce the scope of added tests.

---
In reply to: [185919155](https://github.com/dotnet/roslyn/pull/26545#discussion_r185919155) [](ancestors = 185919155,185918175,185917281)

Reply:
I still believe this is an interesting end user scenario, and tried to add it to an existing unit test to avoid bloating the number of unit tests. Given that you still insist, I will remove the assignment and update the baseline.

---
In reply to: [185922694](https://github.com/dotnet/roslyn/pull/26545#discussion_r185922694) [](ancestors = 185922694,185920922,185919155,185918175,185917281)